diffmatchpatch/diff_test.go

```diff
@@ -998,8 +998,38 @@ func TestDiffText(t *testing.T) {
 }
 
 func TestDiffDelta(t *testing.T) {
+
+	type TestCase struct {
+		Text               string
+		Delta              string
+		ErrorMessagePrefix string
+	}
+
 	dmp := New()
 
+	// Error checking
+	for _, tc := range []TestCase{
+		{"jumps over the lazyx", "=4\t-1\t+ed\t=6\t-3\t+a\t=5\t+old dog", "Delta length (19) smaller than source text length (20)"},
+		{"umps over the lazy", "=4\t-1\t+ed\t=6\t-3\t+a\t=5\t+old dog", "Delta length (19) smaller than source text length (18)"},
+		{"", "+%c3%xy", "invalid URL escape \"%xy\""},
+		{"", "+%c3xy", "invalid UTF-8 token: \"\\xc3xy\""},
+		{"", "a", "Invalid diff operation in DiffFromDelta: a"},
+		{"", "-", "strconv.ParseInt: parsing \"\": invalid syntax"},
+		{"", "--1", "Negative number in DiffFromDelta: -1"},
+		{"", "", ""},
```

Comment:
I agree on what you wrote. The absence of an error must be checked, of course. However, the linked test function handles both error and non-error cases, and I just went with your comment "Error checking" which suggested you wanted to verify only negative cases. As I mentioned in my other commit, it seems that I forgot to refactor this function with the table-driven pattern. I would therefore suggest to drop the "Error checking" comment, and try to put the other cases into the table-driven pattern too. What do you think?